web/modules/custom/dept_core/modules/dept_content_processors/src/Plugin/Filter/AbsToRelUrlsFilter.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Drupal\dept_content_processors\Plugin\Filter;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\dept_core\DepartmentManager;
+use Drupal\filter\FilterProcessResult;
+use Drupal\filter\Plugin\FilterBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a filter to convert absolute URLs in links to relative links
+ * providing the active domain matches the department hostname of the link.
+ *
+ * @Filter(
+ *   id = "dept_abs2rel_url",
+ *   title = @Translation("Department absolute to relative URL"),
+ *   description = @Translation("Converts any absolute URLs for the current department to relative ones."),
+ *   type = Drupal\filter\Plugin\FilterInterface::TYPE_TRANSFORM_REVERSIBLE,
+ * )
+ *
+ * @internal
+ */
+class AbsToRelUrlsFilter extends FilterBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The Department manager.
+   *
+   * @var \Drupal\dept_core\DepartmentManager
+   */
+  protected $departmentManager;
+
+  /**
+   * Constructs a MediaEmbed object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\dept_core\DepartmentManager $department_manager
+   *   The department manager object.
+   *
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, DepartmentManager $department_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
```

Comment:
I'd look at using the $instance pattern which doesn't override the base constructor.

Reply:
Having another attempt at this today but it's functional with just using the dept manager service, so would see the context injection as an enhancement.